x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -285,12 +319,74 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitializat
       try {
         const { analyticsId } = request.params;
-        const results = await context.ml!.mlClient.callAsCurrentUser(
-          'ml.deleteDataFrameAnalytics',
-          {
-            analyticsId,
+        const { deleteTargetIndex, deleteIndexPattern } = request.query;
+        const errorsEncountered = [];
+        let destinationIndex: string | undefined;
+        let deleteTargetIndexAcknowledged: boolean = false;
+        let deleteIndexPatternAcknowledged: boolean = false;
+
+        // Check if analyticsId is valid and get destination index
+        if (deleteTargetIndex || deleteIndexPattern) {
+          try {
+            const dfa = await context.ml!.mlClient.callAsCurrentUser('ml.getDataFrameAnalytics', {
+              analyticsId,
+            });
+            if (Array.isArray(dfa.data_frame_analytics) && dfa.data_frame_analytics.length > 0) {
+              destinationIndex = dfa.data_frame_analytics[0].dest.index;
+            }
+          } catch (e) {
+            return response.customError(wrapError(e));
           }
-        );
+
+          // If user checks box to delete the destinationIndex associated with the job
+          if (destinationIndex && deleteTargetIndex) {
+            // Verify if user has privilege to delete the destination index
+            const userCanDeleteDestIndex = await userCanDeleteIndex(context, destinationIndex);
+            // If user does have privilege to delete the index, then delete the index
+            if (userCanDeleteDestIndex) {
+              try {
+                await context.ml!.mlClient.callAsCurrentUser('indices.delete', {
+                  index: destinationIndex,
+                });
+                deleteTargetIndexAcknowledged = true;
+              } catch (deleteIndexError) {
+                errorsEncountered.push({
+                  msg: `An error occurred deleting destination index ${destinationIndex}`,
+                });
+              }
+            } else {
+              return response.forbidden();
+            }
+          }
+
+          // Delete the index pattern if there's an index pattern that matches the name of dest index
+          if (destinationIndex && deleteIndexPattern) {
+            try {
+              const indexPatternId = await getIndexPatternId(context, destinationIndex);
+              if (indexPatternId) {
+                await deleteIndexPatternById(context, indexPatternId);
+              }
+              deleteIndexPatternAcknowledged = true;
+            } catch (deleteIndexPatternError) {
+              errorsEncountered.push({
+                msg: `An error occurred deleting index pattern ${destinationIndex}`,
+              });
+            }
+          }
+        }
+        // Grab the target index from the data frame analytics job id
+        // Delete the data frame analytics
+        let results = await context.ml!.mlClient.callAsCurrentUser('ml.deleteDataFrameAnalytics', {
```

Comment:
I think this should also be wrapped in a try/catch and a `dataFrameAnalyticsDeleted` boolean returned.
because if this returns an error, we'll be pushed out of this outer `try` and just the error from this request will be returned on line 394.
the information about whether the index and index patterns have been deleted will be lost.
The `errors` array should probably be a list of error objects that have been passed through `wrapError `

Reply:
I changed the array of errors to object with the status after our discussion. There will now be an object for each analyticsJobDeleted, targetIndexDeleted, and indexPatternDeleted. The wrapped error object will be included if an error did happen. If not, error will be omitted.

https://github.com/elastic/kibana/pull/66934/files/17b5fdfe7b7c4413b31dabcfee42abb0614eaa63..807196400c4ef7164e5a44c14075d59d4e3fc218